cli/src/main/java/com/expediagroup/streamplatform/streamregistry/cli/command/delete/EntityClient.java

```diff
@@ -0,0 +1,202 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.cli.command.delete;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.List;
+import java.util.Optional;
+
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+
+import com.apollographql.apollo.ApolloClient;
+import com.apollographql.apollo.api.Operation;
+import com.apollographql.apollo.api.Query;
+
+import picocli.CommandLine.Option;
+
+import com.expediagroup.streamplatform.streamregistry.cli.graphql.ConsumerBindingQuery;
+import com.expediagroup.streamplatform.streamregistry.cli.graphql.ConsumerQuery;
+import com.expediagroup.streamplatform.streamregistry.cli.graphql.DomainQuery;
+import com.expediagroup.streamplatform.streamregistry.cli.graphql.InfrastructureQuery;
+import com.expediagroup.streamplatform.streamregistry.cli.graphql.ProducerBindingQuery;
+import com.expediagroup.streamplatform.streamregistry.cli.graphql.ProducerQuery;
+import com.expediagroup.streamplatform.streamregistry.cli.graphql.SchemaQuery;
+import com.expediagroup.streamplatform.streamregistry.cli.graphql.StreamBindingQuery;
+import com.expediagroup.streamplatform.streamregistry.cli.graphql.StreamQuery;
+import com.expediagroup.streamplatform.streamregistry.cli.graphql.ZoneQuery;
+import com.expediagroup.streamplatform.streamregistry.state.graphql.ApolloExecutor;
+import com.expediagroup.streamplatform.streamregistry.state.graphql.DefaultApolloClientFactory;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.ConsumerBindingKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.ConsumerKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.DomainKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.InfrastructureKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.ProducerBindingKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.ProducerKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.SchemaKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.StreamBindingKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.StreamKey;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity.ZoneKey;
+
+@RequiredArgsConstructor
+class EntityClient {
+  private final ApolloExecutor executor;
+  private final KeyConverter converter;
+
+  @Value
+  static class StreamKeyWithSchemaKey {
+    StreamKey stream;
+    SchemaKey schema;
+  }
+
+  List<DomainKey> domains(String domain) {
```

Comment:
This is an optional change, because I'm being slightly pedantic, but I think a more meaningful method name would help clear up what this method does. I like using verbs so changing this to something like "getDomains" or "getDomainKeys" would be better imo. (Same applies to the rest of the getters in this class)

For example, earlier in the code I was trying to figure out what this line does 
`return unmodifiableList(client.domains(domain))` and it was slightly confusing to me until I reached this point where the method is defined.

Reply:
I generally prefer brevity and think `get` is redundant. I agree it should specify it's getting the keys. Happy to change if it makes it easier to understand.